crypto/kem/kem.c

```diff
@@ -0,0 +1,133 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <openssl/base.h>
+#include <openssl/err.h>
+#include <openssl/mem.h>
+#include <openssl/nid.h>
+
+#include "../fipsmodule/delocate.h"
+#include "../internal.h"
+#include "internal.h"
+
+#define AWSLC_NUM_BUILT_IN_KEMS 1
+
+// TODO(awslc): placeholder OID, replace with the real one when available.
+static const uint8_t kOIDKyber512[] = {0xff, 0xff, 0xff, 0xff};
+
+static const KEM built_in_kems[AWSLC_NUM_BUILT_IN_KEMS] = {
+  {
+    NID_KYBER512,         // kem.nid
+    kOIDKyber512,         // kem.oid
+    sizeof(kOIDKyber512), // kem.oid_len
+    "Kyber512",           // kem.comment
+    800,                  // kem.public_key_len
```

Comment:
Should also reference where these come from.

Reply:
done